iceoryx_posh/include/iceoryx_posh/internal/popo/listener.inl

```diff
@@ -22,39 +22,56 @@ namespace popo
 {
 namespace internal
 {
+template <typename T, typename UserType>
+struct TranslateAndCallTypelessCallback
+{
+    static void call(void* const origin, void* const userType, Listener::GenericCallbackPtr_t underlyingCallback)
+    {
+        reinterpret_cast<typename EventCallback<T, UserType>::Ptr_t>(underlyingCallback)(
+            static_cast<T*>(origin), static_cast<UserType*>(userType));
+    }
+};
+
 template <typename T>
-inline void translateAndCallTypelessCallback(void* const origin, void (*underlyingCallback)(void* const))
+struct TranslateAndCallTypelessCallback<T, internal::NoType_t>
 {
-    reinterpret_cast<void (*)(T* const)>(underlyingCallback)(static_cast<T*>(origin));
-}
+    static void call(void* const origin, void* const userType, Listener::GenericCallbackPtr_t underlyingCallback)
+    {
+        IOX_DISCARD_RESULT(userType);
```

Comment:
can be used this way, but once we have a `IOX_MAYBE_UNUSED` macro, that one would fit better